### PR TITLE
Fixed json cls encoder in trigger_client_event

### LINKED ISSUE #180: Using DjangoJSONEncoder in json.dumps
### Description

Hi,

It might be interesting to use [DjangoJSONEncoder](https://github.com/django/django/blob/7346c288e307e1821e3ceb757d686c9bd879389c/django/core/serializers/json.py#L77) when dumping a dict in trigger_client_event.

https://github.com/adamchainz/django-htmx/blob/825e3493aa5089e7beb44a4c20d92b9e2a0b25f4/src/django_htmx/http.py#L63

```python
response[header] = json.dumps(data, cls=DjangoJSONEncoder)
```

### PR DESCRIPTION
**Pull request by CleitonDeLima**
Fixed #180

### CODE REVIEW

File: tests/test_http.py [hunk 1]
```diff
@@ -99,3 +103,22 @@ def test_success_after_swap(self):
...
+            "datetime": dt,
+            "date": dt.date(),
+        }
+        trigger_client_event(response, "showMessage", params)
+
+        expected = (
+            '{"decimal": "%s", "uuid": "%s", "datetime": "%s", "date": "%s"}'
+        ) % ("9.99", uuid_value, "2022-01-01T10:00:00", "2022-01-01")
+
+        assert expected in response["HX-Trigger"]
```
Comment (adamchainz) [inline tests/test_http.py:124]: Avoid `%` formatting, and where feasible use exact asserts rather than imprecise ones like `in`

**Review by adamchainz — COMMENTED** [review]
Thanks! I'll push a few commits to improve the test, and add docs and changelog note. When you make open source contributions, please try to update the docs and changelog too, in the style of the project. Without such changes your feature will be "secret" and no other users will discover it.

I added some review around the changes to the test below, which I'm taking care of.

**Comment by CleitonDeLima** [comment]
Thanks for the tips @adamchainz , always good to learn!

**Comment by adamchainz** [comment]
No worries, released in version 1.7.0!

**Pull request merged**